### PR TITLE
IAP-IdP-Connect: Connecting IdP (GCP Identity Platform)with IAP(Identity Aware Proxy)

### PR DESCRIPTION
**Pull request by arjunvijaygoogle**
This service programmatically connects IAP (Identity Aware Proxy) to IdP (Identity Platform) in Google Cloud Platform.
By This program, you connect Identity providers (including multi-tenants) with IAP (Identity Aware Proxy) backend services
For example, you connect SAML integrations defined under identity platform for tenant A with backend-service-1 in IAP.
Additionally:
* You can also configure the sign-in page created automatically or custom developed by developers by providing the sign_in_url.
* You can also provide the api key attached to the sign-in page

This functionality doesn't exist as of now with terraform. So creating this as a re-usable asset.

### CODE REVIEW

**Comment by arjunvijaygoogle** [comment]
@agold-rh can you please review this PR

**Comment by arjunvijaygoogle** [comment]
Hi, @aymanfarhat can you please review this PR

File: examples/iap-idp-connect/README.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+For example, you connect SAML integrations defined under identity platform for tenant A with backend-service-1 in IAP.
+Additionally:
+* You can also configure the sign-in page created automatically or custom developed by developers by providing the sign_in_url.
+* You can also provide the api key attached to the sign-in page
+
+
+## Pre-requisite and setting up dependencies
+
+* Install python 3, pip 3 (Make sure Python version is 3.10)
+* Setup application dependencies ():
```
Comment (shakeebshams) [inline examples/iap-idp-connect/README.md:16]: do the parenthesis need to be here?

File: examples/iap-idp-connect/connect_test.py [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+    mock_update_iap_settings.return_value = expected_request
+    # Call the function with mocks
+    with patch(
+        "connect.iap_v1.IdentityAwareProxyAdminServiceClient", return_value=mock_client
+    ):
+        response = update_iap_settings(
+            credentials, project_id, sign_in_url, backend, api_key_arg, tenant_ids
+        )
+
+    # mock_client.assert_called_once_with(credentials=credentials)
```
Comment (shakeebshams) [inline examples/iap-idp-connect/connect_test.py:111]: if unused, might want to remove

File: examples/iap-idp-connect/README.md (hunk 1 above)
Comment (arjunvijaygoogle) [reply examples/iap-idp-connect/README.md]: resolved removed extra paranthesis

File: examples/iap-idp-connect/connect_test.py (hunk 2 above)
Comment (arjunvijaygoogle) [reply examples/iap-idp-connect/connect_test.py]: removed

**Comment by arjunvijaygoogle** [comment]
@shakeebshams I have addressed the code review comments. Can you please let me know if anything else is needed from my end

**Review by agold-rh — APPROVED** [review]

**Comment by arjunvijaygoogle** [comment]
@agold-rh and @shakeebshams Thanks for review and approving this. Please let me know if this can be merged now.

**Comment by agold-rh** [comment]
@arjunvijaygoogle You will have to look at the "Security" tab of your repo to see where Github has identified secrets in your patch. Until they are removed or modified in some way, your patch cannot be merged.

**Pull request closed without merging**